cmd/thanos/receive.go

```diff
@@ -489,20 +497,27 @@ func runReceive(
 	}
 
 	if upload {
-		level.Debug(logger).Log("msg", "upload enabled")
-		if err := dbs.Sync(context.Background()); err != nil {
-			level.Warn(logger).Log("msg", "initial upload failed", "err", err)
+		logger := log.With(logger, "component", "uploader")
+		{
+			level.Info(logger).Log("msg", "upload enabled, starting initial sync")
+			ctx, cancel := context.WithTimeout(context.Background(), uploadTimeout)
+			if err := dbs.Sync(ctx); err != nil {
+				cancel()
+				level.Warn(logger).Log("msg", "initial upload failed", "err", err)
+			}
+			cancel()
+			level.Info(logger).Log("msg", "initial sync done")
 		}
-
 		{
 			// Run the uploader in a loop.
 			ctx, cancel := context.WithCancel(context.Background())
 			g.Add(func() error {
 				return runutil.Repeat(30*time.Second, ctx.Done(), func() error {
+					level.Debug(logger).Log("msg", "recurring upload starting")
 					if err := dbs.Sync(ctx); err != nil {
-						level.Warn(logger).Log("msg", "interval upload failed", "err", err)
+						level.Warn(logger).Log("msg", "recurring upload failed", "err", err)
 					}
-
+					level.Debug(logger).Log("msg", "upload done")
```

Comment:
```suggestion
					level.Debug(logger).Log("msg", "recurring upload done")
```

What about adding `elapsed` time?